bpf-loader-rs/bpf-loader-lib/src/export_event/data_dumper/json.rs

```diff
@@ -0,0 +1,247 @@
+                for i in 0..btf_int.bits / 8 {
+                    // Everything is little-endian, right?
+                    // So we constructed an u64 through shiftings
+                    // Then truncate the corresponding bytes to the type we want
+                    result |= (range[i as usize] as u64) << (i * 8);
+                }
+                let result = match (btf_int.bits, btf_int.encoding) {
+                    (8, BtfIntEncoding::Signed) => json!(result as i8),
+                    (8, _) => json!(result as u8),
+                    (16, BtfIntEncoding::Signed) => json!(result as i16),
+                    (16, _) => json!(result as u16),
+                    (32, BtfIntEncoding::Signed) => json!(result as i32),
+                    (32, _) => json!(result as u32),
+                    (64, BtfIntEncoding::Signed) => json!(result as i64),
+                    #[allow(clippy::unnecessary_cast)]
+                    (64, _) => json!(result as u64),
+                    (a, _) => {
+                        bail!("Unsupported integer length: {} in bits", a);
+                    }
+                };
+
+                Ok(result)
+            }
+        }
+        BtfType::Ptr(_) => {
+            // For pointers, we just interprete them as integers
+            Ok(if range.len() == 4 {
+                json!(u32::from_le_bytes(range[0..4].try_into()?))
+            } else {
+                json!(u64::from_le_bytes(range[0..8].try_into()?))
+            })
+        }
+        BtfType::Array(arr) => {
+            // For c-strings, return a string; For arrays in other types, return a json array
+            let elem_ty = btf.types().get(arr.val_type_id as usize).ok_or_else(|| {
+                anyhow!(
+                    "Invalid element type {} of array {}",
+                    type_id,
+                    arr.val_type_id
+                )
+            })?;
+            let is_c_str = elem_ty.name() == "char";
+            if is_c_str {
+                // Here, this array represents an char[N], which can be interpreted as a string
+                let mut last_idx = 0;
+                while last_idx < range.len() && range[last_idx] != 0 {
+                    last_idx += 1;
+                }
+                let out_str = CStr::from_bytes_with_nul(&range[..=last_idx])?.to_str()?;
+                Ok(json!(out_str))
+            } else {
+                // For non-strings, just create a json array and recursively to fill it
+                let mut result: Vec<Value> = vec![];
+                let elem_size = btf.get_size_of(arr.val_type_id) as usize;
+                for i in 0..arr.nelems as usize {
+                    result.push(dump_to_json(
+                        btf,
+                        arr.val_type_id,
+                        &data[i * elem_size..(i + 1) * elem_size],
+                    )?);
+                }
+                Ok(json!(result))
+            }
```

Comment:
I've thought about it and it doesn't really make much sense. But another piece of advice is good. What do you think, this will reduce nesting and should be easier to read, and importantly, only need a few changes.
```rs
match ty {
    foo => dump_foo(),
    bar => dump_bar(),
    ...
}
```

Reply:
> I've thought about it and it doesn't really make much sense. But another piece of advice is good. What do you think, this will reduce nesting and should be easier to read, and importantly, only need a few changes.
> 
> ```rust
> match ty {
>     foo => dump_foo(),
>     bar => dump_bar(),
>     ...
> }
> ```

OK, I'll apply this